### ISSUE TITLE
Uploading images

### ISSUE THREAD

**Issue by jakubhajek** [thumbor/thumbor#509]
Hi! 
I am playing with uploading image in Thumbor. I executed successfully following command: 

curl -i -H "Content-Type: image/jpeg" -H "Slug: image1.jpg" -XPOST http://localhost:8000/image --data-binary "@/tmp/161072-1-69633-1320414594.jpg"
HTTP/1.1 100 (Continue)

HTTP/1.1 201 Created
Date: Tue, 14 Jul 2015 13:19:25 GMT
Content-Length: 0
Content-Type: text/html; charset=UTF-8
Location: /image/7e4b34a03122458a9376c4094f33d9f8/image1.jpg

Now, I am trying to GET uploaded image with following command: 

curl -i http://localhost:8000/image/7e4b34a03122458a9376c4094f33d9f8/image1.jpg 

and it freezes. 
In the log file I am finding

2015-07-14 15:24:21 tornado.application:ERROR Future exception was never retrieved: Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/tornado-4.2-py2.7-linux-x86_64.egg/tornado/gen.py", line 879, in run
    yielded = self.gen.send(value)
  File "/usr/local/lib/python2.7/dist-packages/thumbor/handlers/image_resource.py", line 34, in check_resource
    mime = BaseEngine.get_mimetype(body)
  File "/usr/local/lib/python2.7/dist-packages/thumbor/engines/__init__.py", line 64, in get_mimetype
    if buffer.startswith('GIF8'):
AttributeError: 'NoneType' object has no attribute 'starts with'

Can you please advise how to fix it ? 

Thank you,

**Comment by guilhermef** [comment]
That's actually a bug, we've changed the way we get the image mime type.

**Comment by jakubhajek** [comment]
Hi, Thank you fro answer - are there any plan where it would be fixed ? 
Thank you,

**Comment by guilhermef** [comment]
I'll take a look, you can use the last tag, it should be OK.

**Comment by jakubhajek** [comment]
Hi, Currently we have running Thumbor 5.0.4 - I have upgraded today and we are experiencing the same issue described at the beginning. 
Appreciate your help on it. 
Thanks,

**Comment by guilhermef** [comment]
That's strange, I'll need more time to dig into it.

**Comment by dhardy92** [comment]
is storage module an external module ?
seams not compatible with new asynchronous API in thumbor 5.0.

**Comment by jakubhajek** [comment]
hello, 
we are using following module:

STORAGE = 'thumbor.storages.file_storage'

just simple filesystem storage. 

Thanks,

**Comment by jakubhajek** [comment]
HI! 
Seems that I have the culprit, I set up

STORAGE_EXPIRATION_SECONDS=0

One of the purpose of installing Thumbor in our infrastructure besides of creating thumbnails - is uploading  lots of images from few external sources through Thumbor instance and store them to our local storage, in the case in regular filesystem. 
My assumption was to store images for ever without any expiration time. 

Would you please elaborate how to properly use that parameter ? 

Thank you,

**Comment by guilhermef** [comment]
I think we could change the [__is_expired](https://github.com/thumbor/thumbor/blob/master/thumbor/storages/file_storage.py#L122) to return False if you set the STORAGE_EXPIRATION_SECONDS to None.

**Comment by jakubhajek** [comment]
Whatever suits you the best - and allow stores images forever without expiration time. 
Thank you,

**Comment by guilhermef** [comment]
@jakubhajek I've made a few changes, now if you set the STORAGE_EXPIRATION_SECONDS to None, it will ignore it. https://github.com/thumbor/thumbor/commit/3c24ff437e468a8654af889384c77f52a850c054

And the error you've reported won't happen anymore.

Do you mind testing with the master ?

**Comment by jakubhajek** [comment]
@guilhermef I don't mind testing Master on our test environment. :) As you suggest I set up STORAGE_EXPIRATION_SECONDS to None and mentioned errors does not longer occur. 
I was able to upload image and than "GET" it from our servers. 

Works like a charm :+1: 

We have to test it before promoting that to production environment and make sure that images are stored forever without expiration. 

Thank you very much for your effort investigation the issue and implementing quick fix. :)
